### PR TITLE
Feat font types

### PR DESCRIPTION
**Pull request by mixcas**
plz check if my font sizes are right ;3

### CODE REVIEW

**Comment by mikeyrayvon** [comment]
how did you determine these rem font sizes @mixcas ? i really believe we should use px. I dont see the benefit of using em or rem at all.

File: css/site.styl [hunk 1]
```diff
@@ -28,6 +28,14 @@ normalize()
 // TYPOGRAPHY
 
 @import 'type'
+@import 'fonts/Din/Din'
```
Comment (varffs) [inline css/site.styl]: include the font files first before type

File: css/site.styl [hunk 2]
```diff
@@ -28,6 +28,14 @@ normalize()
...
+@import 'fonts/AlverataInformal/AlverataInformal'
+
+.font-din
+  font-family: 'Din'
```
Comment (varffs) [inline css/site.styl]: then use a font stack here and a variable `$font-sans` etc here

Comment (varffs) [reply css/site.styl]: also this declaration should be in type.styl

**Review by varffs — CHANGES_REQUESTED** [review]
This wasn't the suggestion I meant. The way it is done in Pindrop is very different. The point of the rems is to set them in the font-sizes() mixin based on the base size. The boilerplate seems to be out of date related to how we have done type for a few projects

**Comment by mikeyrayvon** [comment]
I really disagree with the way its done in pindrop. Type sizes should not be locked to fixed ratios. As the ration between type sizes often changes across viewports. Ex: Basic and small fonts are both be 14px in mobile, and are 16 and 14px in tablet, for the sake of legibility. I believe we should redeclare font sizes in px with mixin for each media query

**Comment by varffs** [comment]
Read the Pindrop file. There is a modulator with minimum sizes. It might not be a good way to do it but they've not asked for any typesize changes on Pindrop for responsive since I added modulation. That made it a drop in solution that very quickly did responsive type sizes, I just had to tweak the modulator setting a little bit by eye. It seems a nice way to scale things down without losing the size relationships. However I concede that it is quite a complex way of doing things and would certainly either need an override or maybe not be at a good way at all.

**Comment by varffs** [comment]
I think we should move to 1 of 2 approaches for font styles. 

We can either keep down this route with the mixin, but make the adjustments we decided on i.e. removing mapping of sizes to tags so no `$font-size-h1` set manually to an `h3` because thats the designs. If we are using mixin approach then I favor and like the modulator as it is a very rapid way to have a mobile type size setup that works.

The other approach is to move away from the mixin for sizes and map out all the different font styles used in a design. Then all these styles are declared as mapped to classes and all the markup is reset (other than the base HTML declaration) And we just have to remember to redeclare as needed on the responsive breakpoints. Once we start doing mobile first properly this will be cleaner than it is now.

Advantage of first is speed once configured, advantage of second is more fine tune customization as well as less duplicate and unneeded generated css.

I think I would advocate having a quality mixin in the boiler that we can use to prototype or one certain projects, but that we should move towards the second option in the future. The first step of the second option is to fix the mobile first approach

File: css/type.styl [hunk 3]
```diff
@@ -93,36 +96,40 @@ a:active,
...
+    else
+      font-size: ($size-basic + 1px)
+
+  .font-size-heading
+    if $size-heading > ($size-basic + 1px)
+      font-size: $size-heading
+    else
+      font-size: ($size-basic + 1px)
+
+  .font-large
```
Comment (varffs) [inline css/type.styl]: I think these should be `.font-size-large` etc

File: page-info.php [hunk 4]
```diff
@@ -19,7 +19,7 @@
 ?>
       <section class="grid-row">
         <div class="grid-item item-m-4">
-          <h2>About</h2>
+          <h2 class="font-heading font-uppercase font-size-heading">About</h2>
```
Comment (varffs) [inline page-info.php]: Its kinda annoying that `.font-heading` is so close to the size class. I wonder if it should be `.font-family-heading` or `.font-typeface-heading` or `.font-file-heading` ?

**Review by varffs — CHANGES_REQUESTED** [review]

**Comment by mixcas** [comment]
I decided to go with the modulator mixing approach for now since we don't have mobile designs yet.

Maybe we should improve the starter-pack on this part, let's do it when @mikeyrayvon gets back.

File: page-info.php (hunk 4 above)
Comment (mixcas) [reply page-info.php]: imma fix it to `.font-family-din`

Comment (mixcas) [reply page-info.php]: ugh, nvm i stopped like it

File: css/type.styl (hunk 3 above)
Comment (varffs) [reply css/type.styl]: @mixcas and this?

**Comment by mikeyrayvon** [comment]
My feeling is that the mixin should be configured to never or almost never be modified between projects. I fear that the modulator approach youre using now will almost always need to be modified in the mixin. because the else condition of `basic + 1` just doesnt work for every project (or even any project?). what is the use case proof for this approach? The mixin that is defined in the static starter pack would allow for either modulator or px perfect font size class declarations. but you would do the math in the query instead of the mixin.  i think thats more flexible / future proof

**Review by varffs — APPROVED** [review]

**Comment by varffs** [comment]
I see your point Mikey but for the sake of this nightmare PR I'm just going to merge it. The classes we are using are correct so we can refactor the stylus later esp if we get a different design for mobile or whatever. Also think we should do a full stylus refactor for the starter pack soon

**Pull request merged**